src/Combinatorics/IntegerCombinatorics/partitions.jl

```diff
@@ -269,10 +209,8 @@ function partitions(n::IntegerUnion)
     push!(P, partition(d[1:k]))
   end
   return P
-
 end
 
-
 @doc raw"""
     ascending_partitions(n::IntegerUnion; algorithm::Symbol=:ks)
```

Comment:
I'm confused about this function. Reading the doc string, it looks like this was just an experiment: A paper claims that generating ascending partitions is faster than descending ones (which is the usual convention), but this appears not to be the case with the implementation. So, why is this function here (and even exported!)?

If I understand the situation correctly and we want to make all implemented algorithms available to the user, I would suggest to give `partitions` an optional argument to choose an algorithm. If it is one that generates ascending partitions, we can turn them around.

There was also some discussion https://github.com/oscar-system/Oscar.jl/pull/1935#discussion_r1131794777 about the timings. For what it's worth, here are mine:
```
julia> @btime partitions(Int8(90));
  8.120 s (56634195 allocations: 4.53 GiB)

julia> @btime ascending_partitions(Int8(90), algorithm=:ks);
  20.021 s (56634195 allocations: 4.53 GiB)

julia> @btime ascending_partitions(Int8(90), algorithm=:m);
  20.054 s (56634195 allocations: 4.53 GiB)
```

Reply:
I remember: compositions need to be ported as well and then I wanted to move ascending_partitions to ascending_compositions in the compositions file. I have to check my code where I stopped with this.

Reply:
https://github.com/ulthiel/Oscar.jl/tree/ut/compositions was my starting point but I forgot what I did.

Reply:
Okay, so for now I would suggest to stop exporting `ascending_partitions` (and maybe remove it from the documentation).

Reply:
and maybe just leave it unexported in experimental for the time being?